### ISSUE TITLE
PS5 4.3

### ISSUE THREAD

**Issue by rebeccajohnson88** [rebeccajohnson88/qss20_slides_activities#57]
## 4.3 Predicting disparities (10 points)

**A.** Read in the raw `acs_wmath` data again (so loses the variables you created above)

**B.** Construct a binary indicator variable for male score > female score  for each year - for full credit, do so without repeating the difference code for each of the four years: name these according to following convention: `outcome_male_higher_female_year` where year is 2013, 2015, 2017, or 2019 (e.g., 2013: `outcome_male_higher_female_2013`). After this, remove the raw math scores as columns in the data (so filter out any column with the word math)

**C.** Melt the data (`acs_wmath`) to long where instead of wide years, years are repeated within state; the acs vars will also be repeated since we only pulled one year 

**D.** Split into train-test split at state level (so all years in same state -> either all in train or all test). Randomize 35 states to train; 15 states in test. 

**E.** Normalize the features to mean 0, variance 1 and estimate a **decision tree with a max depth of 5**. 

**F.** Interpret the feature importances

**G.** Evaluate the precision and recall of that model in the test set states without using the `score`, `precision`, or `recall` functions in sklearn. Briefly interpret: compared to our class example (a high-dimensional feature matrix of yelp reviews with ~15000 observations), why do you think our models perform worse for this set of data/predictors?

Resources: 
-https://github.com/rebeccajohnson88/qss20_slides_activities/blob/main/activities/w22_activities/solutions/09_binaryclassification_activity_solutions.ipynb

- Feature normalization: https://scikit-learn.org/stable/modules/preprocessing.html

- Definition of precision and recall: https://en.wikipedia.org/wiki/Precision_and_recall

**Some outputs (fine for your model-related outputs to differ due to train-test split**:

Head of long-format data (selected cols)

![image](https://user-images.githubusercontent.com/21369316/156242917-f44a31f9-412f-4b94-8291-3cc5978afe86.png)

Top features (fine for yours to differ):

![image](https://user-images.githubusercontent.com/21369316/156243094-e0ea7d4c-7aff-4008-97bc-845d41288bdb.png)

Precision and recall (fine for yours to differ but will generally be pretty low):

![image](https://user-images.githubusercontent.com/21369316/156243386-c74ab345-2434-4585-9781-4df7f195a5e0.png)

**Comment by MarcoAllen1** [comment]
are state and median income the columns we should be keeping in our melt? As in it is okay to lose all other columns?

**Comment by MarcoAllen1** [comment]
Assume we should normalize both X_train but also initialize a new scaler for X_test to normalize that as well when checking output? Is it correct to say that a unique scaler would need to be initialized for each? Or could we use the same one?

**Comment by rebeccajohnson88** [comment]
> Assume we should normalize both X_train but also initialize a new scaler for X_test to normalize that as well when checking output? Is it correct to say that a unique scaler would need to be initialized for each? Or could we use the same one?

Either is fine pointswise but literature recommends using the training set scaler to transform the test set for reasons discussed here: https://stats.stackexchange.com/questions/495357/why-do-we-normalize-test-data-on-the-parameters-of-the-training-data

**Comment by rebeccajohnson88** [comment]
Question: `I am confused about the instructions for the melt in part 4… can you further explain what you mean?`

Response: based on the sample output (reposted here), you should reshape the data from "wide format" where each state is a row and we have separate columns for each year to "long format" where states are repeated four times: once for each year in the data (2013, 2015, 2017, 2019)

![image](https://user-images.githubusercontent.com/21369316/157962966-707a79b0-cdad-4ddd-b2c4-cc48bd8af115.png)
